coordinator/tributary/src/mempool.rs

```diff
@@ -118,27 +175,31 @@ impl<D: Db, T: Transaction> Mempool<D, T> {
   pub(crate) fn block(
     &mut self,
     blockchain_next_nonces: &HashMap<<Ristretto as Ciphersuite>::G, u32>,
-  ) -> Vec<T> {
+  ) -> Vec<Transaction<T>> {
     let mut res = vec![];
     for hash in self.txs.keys().cloned().collect::<Vec<_>>() {
       let tx = &self.txs[&hash];
+
       // Verify this hasn't gone stale
-      match tx.kind() {
-        TransactionKind::Signed(Signed { signer, nonce, .. }) => {
-          if blockchain_next_nonces[signer] > *nonce {
-            self.remove(&hash);
-            continue;
+      if let Transaction::Application(tx) =  tx {
+        match tx.kind() {
+          TransactionKind::Signed(Signed { signer, nonce, .. }) => {
+            if blockchain_next_nonces[signer] > *nonce {
+              self.remove(&hash);
+              continue;
+            }
           }
+          _ => panic!("non-signed transaction entered mempool"),
         }
-        _ => panic!("non-signed transaction entered mempool"),
       }
 
       // Since this TX isn't stale, include it
+      // TODO: This also includes non-signed Tendermint Evidence txs. should it?
```

Comment:
Yes, yet preferably in a seperate vec so we can return `unsigned || signed`, and enforce that ordering (unsigned before signed, just as we currently have provided before signed).

Reply:
why do we have that ordering? can't a block contain all 3 types in the same time? or you mean the tx ordering within the same block?

Reply:
Within a block, ideally the transactions are ordered

Provided
Unsigned
Signed